src/components/Main/Main.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { MainProps } from '../../types/MainProps';
+import { TodoItem } from '../TodoItem';
+import { Filter } from '../../types/Filter';
+
+export const Main: React.FC<MainProps> = ({
+  todos,
+  filterBy,
+  tempTodo,
+  handleTodoDelete,
+  deletingCompleted,
+}) => {
+  const visibleTodos = todos.filter(todo => {
+    switch (filterBy) {
+      case Filter.active:
+        return !todo.completed;
+      case Filter.completed:
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {visibleTodos.map(todo => (
+        <TodoItem
+          key={todo.id}
+          todo={todo}
+          tempTodo={tempTodo}
+          onDelete={handleTodoDelete}
+          deletingCompleted={deletingCompleted}
+        />
+      ))}
+
+      {tempTodo && (
+        <TodoItem
+          todo={tempTodo}
+          tempTodo={tempTodo}
```

Comment:
looks strange, let's just pass into TodoItem optional prop isLoading, so here you need to add
```suggestion
          todo={tempTodo}
          isLoading
```

Reply:
Oh, that is right we can remove one prop, thanks for that.